### PR TITLE
[kube-prometheus-stack] Migrate to chart v2

### PR DESCRIPTION
**Pull request by mrueg**
#### What this PR does / why we need it:
Initial attempt to move to chart v2, since helm v2 is running out of support.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by vsliouniaev** [comment]
We're currently stuck on Helm 2 and are relying on this chart. Could we delay doing this upgrade until there is need to break this compatability?

**Comment by mrueg** [comment]
> We're currently stuck on Helm 2 and are relying on this chart. Could we delay doing this upgrade until there is need to break this compatability?

Grafana's chart has migrated over to helm3 already, I'm not sure if delaying means that we will be stuck on the current grafana?

**Comment by r3kzi** [comment]
> We're currently stuck on Helm 2 and are relying on this chart. Could we delay doing this upgrade until there is need to break this compatability?

After November 13, 2020 no further Helm v2 releases (even for security patches) will be published. So IMHO we should not support Helm v2 features after this date.

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -33,3 +33,17 @@ annotations:
...
+  repository: https://charts.helm.sh/stable
+  condition: kubeStateMetrics.enabled
+- name: prometheus-node-exporter
+  version: "1.12.*"
+  repository: https://prometheus-community.github.io/helm-charts
+  condition: nodeExporter.enabled
+- name: grafana
+  version: "5.8.*"
+  repository: https://grafana.github.io/helm-charts
+  condition: grafana.enabled
```
Comment (torstenwalter) [inline charts/kube-prometheus-stack/Chart.yaml:49]: I like it that you keep dependencies at the same version, while making the switch to helm 3.

**Comment by torstenwalter** [comment]
I can only agree to what @r3kzi states. As helm 2 will no longer be supported in a couple of days I see no reason of holding back this change.

It's also the correct chart to start with as there should not be any issue with a helm 3 chart depening on a helm 2 chart.

**Comment by torstenwalter** [comment]
@mrueg can you resolve conflicts?

**Comment by mrueg** [comment]
@torstenwalter Thanks, I fixed the conflicts.

**Review by torstenwalter — APPROVED** [review]

**Comment by torstenwalter** [comment]
I approved this as technically his changes look correct. For this to be merged a maintainer of the chart needs to approve and merge it.

File: charts/kube-prometheus-stack/Chart.yaml (hunk 1 above)
Comment (mrueg) [reply charts/kube-prometheus-stack/Chart.yaml:49]: I plan to do a follow-up PR bumping grafana to a later version.

**Review by vsliouniaev — APPROVED** [review]

File: charts/kube-prometheus-stack/README.md [hunk 2]
```diff
@@ -163,26 +161,6 @@ You can read more information on how to add firewall rules for the GKE control p
 
 Alternatively, you can disable the hooks by setting `prometheusOperator.admissionWebhooks.enabled=false`.
 
-### Helm fails to create CRDs
```
Comment (vsliouniaev) [inline charts/kube-prometheus-stack/README.md:166]: [Helm 3 will only install CRDs and not perform any updates on them afterwards, any changes to CRDs still require someone to maintain them outside of helm](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#method-1-let-helm-do-it-for-you). Do you think we should move this section into the upgrade instructions?

**Review by vsliouniaev — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 1 above)
Comment (scottrigby) [reply charts/kube-prometheus-stack/Chart.yaml:49]: Only thing is, updating to the newer version of Grafana is the only argument I can see for moving this chart to v2 now.

File: charts/kube-prometheus-stack/README.md (hunk 2 above)
Comment (scottrigby) [reply charts/kube-prometheus-stack/README.md:166]: Yes we should, if there were past versions of the CRDs, in the upgrading section for the chart version where the new CRD versions were introduced, we should note that they need to be manually upgraded to that newer version 👍

Comment (r3kzi) [reply charts/kube-prometheus-stack/README.md:166]: @mrueg can you accept the requested changes so we can proceed with merging the Pull Request?

**Review by vsliouniaev — APPROVED** [review]

**Comment by vsliouniaev** [comment]
I'd be happy if we can update the Readme in a follow-up PR with a general statement about CRD behaviour.

File: charts/kube-prometheus-stack/README.md (hunk 2 above)
Comment (mrueg) [reply charts/kube-prometheus-stack/README.md:166]: @r3kzi there's nothing I can accept from a Github UI, I added a paragraph to the update piece linking the doc that @vsliouniaev  mentioned.

**Comment by mrueg** [comment]
@scottrigby @vsliouniaev @r3kzi would you agree that a CRD change should result in a major version bump of the chart as a signal? Maybe that's something we can include as well?

**Comment by vsliouniaev** [comment]
I think that's a really good way to do it

**Review by vsliouniaev — APPROVED** [review]

**Pull request merged**